includes/handlers/class-webmention-handler-mf2.php

```diff
@@ -0,0 +1,650 @@
+	 *
+	 * @var string
+	 */
+	protected $slug = 'mf2';
+
+	/**
+	 * Takes a request object and parses it.
+	 *
+	 * @param Webmention_Request $request Request Object.
+	 * @param Webmention_Item $item A Parsed Item. If null, a new one will be created.
+	 * @return WP_Error|true Return error or true if successful.
+	 */
+	public function parse( Webmention_Request $request ) {
+		$dom = clone $request->get_domdocument();
+		if ( ! class_exists( '\Webmention\Mf2\Parser' ) ) {
+			require_once plugin_dir_path( __FILE__ ) . '../../libraries/mf2/Mf2/Parser.php';
+		}
+
+		$url      = $request->get_url();
+		$parser   = new Webmention\Mf2\Parser( $dom, $url );
+		$mf_array = $parser->parse();
+
+		// Attempts to remove everything but the representative item.
+		$mf_array = $this->get_representative_item( $mf_array, $url );
+		$return   = $this->add_properties( $mf_array );
+		$this->webmention_item->set_url( $url ); // If there is no URL property then use the retrieved URL.
+		return is_wp_error( $return ) ? $return : true;
+	}
+
+
+	/**
+	 * Takes mf2 json and generates a Webmention Item.
+	 *
+	 * @param array $mf_array JSON Array of Parsed Microformats.
+	 * @return WP_Error|true Return error or true if successful.
+	 */
+	public function add_properties( $mf_array ) {
+
+		// Only store the raw representative item and discard other information.
+		$this->webmention_item->set_raw( $mf_array );
+
+		// Retrieve time properties if available.
+		$this->webmention_item->set_published( $this->get_datetime_property( 'published', $mf_array ) );
+		$this->webmention_item->set_updated( $this->get_datetime_property( 'updated', $mf_array ) );
+
+		$this->webmention_item->set_url( $this->get_plaintext( $mf_array, 'url' ) );
+		$this->webmention_item->set_author( $this->get_author( $mf_array ) );
+		$this->webmention_item->set_category( $this->get_property( $mf_array, 'category' ) );
+		$this->webmention_item->set_syndication( $this->get_plaintext( $mf_array, 'syndication' ) );
+
+		// Sometimes the featured image is stored in featured. Otherwise try photo.
+		$this->webmention_item->set_photo( $this->get_plaintext( $mf_array, 'featured' ) );
+		$this->webmention_item->set_photo( $this->get_plaintext( $mf_array, 'photo' ) );
+
+		$this->webmention_item->set_location( $this->get_location( $mf_array ) );
+
+		$this->webmention_item->set_summary( $this->get_html( $mf_array, 'summary' ) );
+		$this->webmention_item->set_content( $this->get_html( $mf_array, 'content' ) );
+		$this->webmention_item->set_response_type( $this->response_type_discovery( $mf_array ) );
+
+		return true;
+	}
+
+	/**
+	 * Returns the response type based on parsed microformats.
+	 * Adds some experimental response types.
+	 * @link http://ptd.spec.indieweb.org/#response-algorithm
+	 *
+	 * @param array $mf_array JSON Array of Parsed Microformats.
+	 * @return string Response Type. Default 'mention'.
+	 */
+	public function response_type_discovery( $mf_array ) {
```

Comment:
this is a lot of ifs... can we simplify this by iterating through arrays or so? This also does not use the comment type object at all.

Reply:
I would like to see that the logic is in the comment-type object, so that it is possible to be used in different contexts (JSON-LD, OGP, ...).

Reply:
Example: you register a comment-type with the pattern to discover it.

Reply:
I originally thought that the comment object would make sense for this, as I commented...however... RSVP is a data property which might need type validation. I tried to fix tagging by making it only work if the person-tag is of the home url...people may tag multiple people in the same post... 

Let me think about how it could be best done, because using the comment object was my original plan. It would probably have to be a function in the comment object that passed through some information.

Reply:
The problem is....with RSVP for example, an RSVP has an in-reply-to property...so if this is not done in a strict order, it would declare itself a reply, not an RSVP.

Reply:
we can reuqire an order, no problem!

Reply:
I'm still having trouble with this, much as I'd like to do it, because of the complexity of the problem. I think, to address your issues, I could move some of it into the handler class...